src/main/java/it/pagopa/ecommerce/commons/utils/ConfidentialDataManager.java

```diff
@@ -74,114 +88,155 @@ public interface ConfidentialData {
-     * Descrypts the data to string without conversion. Prefer {@link ConfidentialDataManager#decrypt(Confidential, Function)} to this.
+     * <p>
+     * Decrypts the data to string without conversion. Prefer {@link ConfidentialDataManager#decrypt(Confidential, Function)} to this.
+     * </p>
+     * <br>
+     * <p>
+     * In case of error the returned {@link Mono} contains a {@link ConfidentialDataException} wrapping one of the following exceptions:
+     * <ul>
+     *      <li>InvalidAlgorithmParameterException See {@link javax.crypto.Cipher#doFinal(byte[])}</li>
+     *      <li>NoSuchPaddingException See {@link javax.crypto.Cipher#doFinal(byte[])}</li>
+     *      <li>IllegalBlockSizeException See {@link javax.crypto.Cipher#doFinal(byte[])}</li>
+     *      <li>NoSuchAlgorithmException See {@link javax.crypto.Cipher#doFinal(byte[])}</li>
+     *      <li>BadPaddingException See {@link javax.crypto.Cipher#doFinal(byte[])}</li>
+     *      <li>InvalidKeyException See {@link javax.crypto.Cipher#doFinal(byte[])}</li>
+     * </ul>
+     *
      * @param data the data to be decrypted
-     * @return a {@code T} instance built from the encrypted data
+     * @return a {@code Mono<T>} instance built from the encrypted data
      * @param <T> the type of the object to be returned
-     * @throws InvalidAlgorithmParameterException See {@link javax.crypto.Cipher#doFinal(byte[])}
-     * @throws NoSuchPaddingException See {@link javax.crypto.Cipher#doFinal(byte[])}
-     * @throws IllegalBlockSizeException See {@link javax.crypto.Cipher#doFinal(byte[])}
-     * @throws NoSuchAlgorithmException See {@link javax.crypto.Cipher#doFinal(byte[])}
-     * @throws BadPaddingException See {@link javax.crypto.Cipher#doFinal(byte[])}
-     * @throws InvalidKeyException See {@link javax.crypto.Cipher#doFinal(byte[])}
      */
     @Nonnull
-    public <T extends ConfidentialData> String decrypt(Confidential<T> data) throws NoSuchPaddingException, NoSuchAlgorithmException,
-            InvalidAlgorithmParameterException, InvalidKeyException,
-            BadPaddingException, IllegalBlockSizeException {
+    public <T extends ConfidentialData> Mono<String> decrypt(Confidential<T> data) {
 
         return switch (data.confidentialMetadata()) {
-            case AESMetadata aesMetadata -> this.aesCipher.decrypt(aesMetadata, data.opaqueData());
+            case AESMetadata aesMetadata -> {
+                try {
+                    yield Mono.just(this.aesCipher.decrypt(aesMetadata, data.opaqueData()));
+                } catch (NoSuchPaddingException | NoSuchAlgorithmException | InvalidAlgorithmParameterException |
+                         InvalidKeyException | BadPaddingException | IllegalBlockSizeException e) {
+                    throw new ConfidentialDataException(e);
+                }
+            }
+            case PersonalDataVaultMetadata ignored -> this.personalDataVaultClient.findPiiUsingGET(data.opaqueData()).map(PiiResourceDto::getPii);
```

Comment:
Catch here the `WebClientResponseException` that could be thrown by the underling client and wrap it to ConfidentialDataException

Reply:
Fixed in [4453ed2](https://github.com/pagopa/pagopa-ecommerce-commons/pull/57/commits/4453ed21820b9d896945f4d9cd4df8bdea37f49b)